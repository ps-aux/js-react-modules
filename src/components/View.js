import React from 'react'
import { addClassName } from '../utils'

const column = {
    display: 'flex',
    flexDirection: 'column'
}

const row = {
    display: 'flex',
    flexDirection: 'row'
}

const cn = addClassName('view')

const View = ({children, className, horizontal = false, ...rest}) =>
    <div style={horizontal ? row : column} className={cn(className)} {...rest}>
        {children}
    </div>

export default View
