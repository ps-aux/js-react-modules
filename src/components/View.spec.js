import React from 'react'
import renderer from 'react-test-renderer'
import View from './View'

describe('View renders', () => {
    it('children', () => {
        const comp = renderer.create(
            <View>
                <div>Child 1</div>
                <div>Child 1</div>
            </View>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('additional className', () => {
        const comp = renderer.create(
            <View className="additional">any</View>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('horizontaly when set', () => {
        const comp = renderer.create(
            <View horizontal>
                <div>1</div>
                <div>2</div>
            </View>)

        const tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
    })

})
