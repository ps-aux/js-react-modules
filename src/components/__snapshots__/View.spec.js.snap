// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View renders additional className 1`] = `
<div
  className="view additional"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  any
</div>
`;

exports[`View renders children 1`] = `
<div
  className="view"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div>
    Child 1
  </div>
  <div>
    Child 1
  </div>
</div>
`;

exports[`View renders horizontaly when set 1`] = `
<div
  className="view"
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;
